common/components/Header/components/NetworkSelector/NetworkSelector.tsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import translate, { translateRaw } from 'translations';
+import { DropdownShell } from 'components/ui';
+import NetworkOption from './NetworkOption';
+import {
+  TChangeNodeIntent,
+  changeNodeIntent,
+  TChangeNetworkIntent,
+  changeNetworkIntent
+} from 'actions/config';
+import {
+  getNodeId,
+  getNodeConfig,
+  getNetworkConfig,
+  getAllNodes,
+  getAllNetworkConfigs
+} from 'selectors/config';
+import { isAutoNodeConfig } from 'libs/nodes';
+import { NodeConfig } from 'types/node';
+import { NetworkConfig } from 'types/network';
+import { AppState } from 'reducers';
+import './NetworkSelector.scss';
+
+const CORE_NETWORKS = ['ETH', 'ETC', 'Ropsten', 'Kovan', 'Rinkeby'];
+
+interface OwnProps {
+  openCustomNodeModal(): void;
+}
+
+interface StateProps {
+  nodeSelection: string;
+  node: NodeConfig;
+  network: NetworkConfig;
+  allNodes: { [key: string]: NodeConfig };
```

Comment:
We really should just make a generic for this, like https://github.com/MyCryptoHQ/shepherd/blob/master/src/types/index.d.ts#L41